tree/ntupleutil/v7/src/RNTupleInspector.cxx

```diff
@@ -22,114 +22,110 @@
+   : fPageSource(std::move(pageSource))
 {
+   fPageSource->Attach();
    auto descriptorGuard = fPageSource->GetSharedDescriptorGuard();
-   int compressionSettings = -1;
-   std::uint64_t compressedSize = 0;
-   std::uint64_t uncompressedSize = 0;
-
-   for (const auto &clusterDescriptor : descriptorGuard->GetClusterIterable()) {
-      compressedSize += clusterDescriptor.GetBytesOnStorage();
-
-      if (compressionSettings == -1) {
-         compressionSettings = clusterDescriptor.GetColumnRange(0).fCompressionSettings;
-      }
-   }
-
-   for (uint64_t colId = 0; colId < descriptorGuard->GetNPhysicalColumns(); ++colId) {
-      const ROOT::Experimental::RColumnDescriptor &colDescriptor = descriptorGuard->GetColumnDescriptor(colId);
-
-      uint64_t elemSize =
-         ROOT::Experimental::Detail::RColumnElementBase::Generate(colDescriptor.GetModel().GetType())->GetSize();
-
-      uint64_t nElems = descriptorGuard->GetNElements(colId);
-      uncompressedSize += nElems * elemSize;
-   }
-
-   fCompressionSettings = compressionSettings;
-   fCompressedSize = compressedSize;
-   fUncompressedSize = uncompressedSize;
+   fDescriptor = descriptorGuard->Clone();
 }
 
-void ROOT::Experimental::RNTupleInspector::CollectColumnData()
+void ROOT::Experimental::RNTupleInspector::CollectColumnInfo()
 {
-   auto descriptorGuard = fPageSource->GetSharedDescriptorGuard();
-
-   for (DescriptorId_t colId = 0; colId < descriptorGuard->GetNPhysicalColumns(); ++colId) {
-      const ROOT::Experimental::RColumnDescriptor &colDescriptor = descriptorGuard->GetColumnDescriptor(colId);
-
+   for (DescriptorId_t colId = 0; colId < fDescriptor->GetNPhysicalColumns(); ++colId) {
       RColumnInfo info;
-      info.fPhysicalColumnId = colId;
-      info.fLogicalColumnId = colDescriptor.GetLogicalId();
-      info.fFieldId = colDescriptor.GetFieldId();
-      info.fIndex = colDescriptor.GetIndex();
-      info.fType = colDescriptor.GetModel().GetType();
+      info.fColumnDescriptor = &(fDescriptor->GetColumnDescriptor(colId));
 
       // We generate the default memory representation for the given column type in order
       // to report the size _in memory_ of column elements.
-      uint64_t elemSize =
-         ROOT::Experimental::Detail::RColumnElementBase::Generate(colDescriptor.GetModel().GetType())->GetSize();
-      info.fElementSize = elemSize;
+      info.fElementSize = ROOT::Experimental::Detail::RColumnElementBase::Generate(info.GetType())->GetSize();
+
+      for (const auto &clusterDescriptor : fDescriptor->GetClusterIterable()) {
+         if (!clusterDescriptor.ContainsColumn(colId)) {
+            continue;
+         }
 
-      for (const auto &clusterDescriptor : descriptorGuard->GetClusterIterable()) {
          auto columnRange = clusterDescriptor.GetColumnRange(colId);
          info.fNElements += columnRange.fNElements;
 
+         if (fCompressionSettings == -1) {
+            fCompressionSettings = columnRange.fCompressionSettings;
```

Comment:
Yeah good point! To add, if the compression across columns might vary in the future, I don't think it would make much sense to keep this method around at all.